### PR TITLE
Remove old job code

### PR DESCRIPTION
**Pull request by josenavas**
Depends on #2044 so review/merge that one first.

Removes the old job.py code and its tests.

### CODE REVIEW

File: qiita_db/support_files/patches/47.sql [hunk 1]
```diff
@@ -0,0 +1,115 @@
+-- Jan 5, 2017
+-- Move the analysis to the plugin system. This is a major rewrite of the
+-- database backend that supports the analysis pipeline.
+-- After exploring the data on the database, we realized that
+-- there are a lot of inconsistencies in the data. Unfortunately, this
+-- makes the process of trasnferring the data from the old structure
```
Comment (mortonjt) [inline qiita_db/support_files/patches/47.sql]: trasnferring -> transferring

File: qiita_db/support_files/patches/47.sql [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+-- to the new one a bit more challenging, as we will need to handle
+-- different special cases. Furthermore, all the information needed is not
+-- present in the database, since it requires checking BIOM files. Due to these
+-- reason, the vast majority of the data transfer is done in the python patch
+-- 47.py
+
+-- In this file we are just creating the new data structures. The old
+-- datastructure will be dropped in the python patch once all data has been
```
Comment (mortonjt) [inline qiita_db/support_files/patches/47.sql]: datastructure -> data structure

File: qiita_db/support_files/patches/47.sql [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+-- table. We will do that on the patch 47.py using the
+-- purge_filepaths function, as it will make sure that those files are
+-- not used anywhere else
+DELETE FROM qiita.job_results_filepath WHERE job_id IN (
+    SELECT job_id FROM qiita.job J WHERE NOT EXISTS (
+        SELECT * FROM qiita.analysis_job AJ WHERE J.job_id = AJ.job_id));
+DELETE FROM qiita.job J WHERE NOT EXISTS (
+    SELECT * FROM qiita.analysis_job AJ WHERE J.job_id = AJ.job_id);
+
+-- In the analysis pipeline, an artifact can have mutliple datatypes
```
Comment (mortonjt) [inline qiita_db/support_files/patches/47.sql]: mutliple -> multiple

File: qiita_db/support_files/patches/47.sql [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+-- (e.g. procrustes). Allow this by creating a new data_type being "multiomic"
+INSERT INTO qiita.data_type (data_type) VALUES ('Multiomic');
+
+
+-- The valdiate command from BIOM will have an extra parameter, analysis
```
Comment (mortonjt) [inline qiita_db/support_files/patches/47.sql]: valdiate -> validate

**Comment by mortonjt** [comment]
Can you provide a little more context on what's going on here?  Why is all of this code being removed? More details will make it easier for review

**Comment by josenavas** [comment]
Hey @mortonjt 

Thanks for reviewing, but make sure to read the comments of the PRs first. This PR depends on a couple other PR that need to be reviewed/merged first, so we have an incremental refactor.

Answering your question, this PR is part of of the Analysis refactor in which we move away from having hardcoded commands to use the plugin structure.

For your reference, the first PR that needs to be reviewed and merged is https://github.com/biocore/qiita/pull/2040

**Comment by josenavas** [comment]
Quick one!! This is just removing the "job" class as now the analysis goes through the "ProcessingJob" object 😄  love removing code!

**Pull request merged**